api/v1/cluster_webhook.go

```diff
@@ -799,6 +800,56 @@ func (r *Cluster) validateBootstrapRecoverySource() field.ErrorList {
 	return result
 }
 
+// validateBootstrapRecoveryDataSource is used to ensure that the data
+// source is correctly defined
+func (r *Cluster) validateBootstrapRecoveryDataSource() field.ErrorList {
```

Comment:
As we already discussed, we should also validate `apiGroup` and `kind` to be of resources we can actually handle, given the current implementation is transparently passing it down to the PVC's [DataSource](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L496), these should be: `PersistentVolumeClaim` and `snapshot.storage.k8s.io/VolumeSnapshot`